### ISSUE TITLE
Questions about MIM and K list value

### ISSUE THREAD

**Issue by lu77777777** [zhengzangw/PCS-FUDA#7]
Thank you so much for sharing this terrific project.

I was confused about Mutual Information Maximization(MIM) for a long time. I guess the MIM  in your paper is just like maximizing the mutual information between input and output?  And in your code, there is a function called MomentumSoftmax. I am not sure why use this function to achieve MIM.  I also monitor the condentmax and entmin loss. It seems just slightly change during the training stage.

And I am not sure how to set the K list value for clustering? Is this value depend on the category’s number of classifications?

Thank you so much

**Comment by zhengzangw** [comment]
Your understanding is correct. In our code, the MIM loss is composed of two losses: semi/tgt-entmin and semi/tgt-condentmax, which refers to the two parts in equation (12) of the paper. The MomentumSoftmax is to keep track of global probability as stated in the last paragraph of Appendix A.

The K list value can be changed by modifying the JSON file. In the "clus" property, "n_k" means the replication time of the "k" list. "k" means different k values you want to use. See the example config for more details.

**Comment by lu77777777** [comment]
Thank you for your reply. I tried to apply this project to a data-limited dataset. Because of limited data, the base encoder cannot be pretraining, so the performance isn't so good. Do you have any suggestions?

**Comment by zhengzangw** [comment]
My suggestion is to use a large dataset pretrained model (e.g., ImageNet-1k). We find that initial prototype quality is very important, so a pretrained model is needed for a good performance. Another potential solution is to train your model with only cross-entropy first and then with our method.

**Comment by lu77777777** [comment]
Thank you so much

**Comment by lu77777777** [comment]
Thanks a lot. I tried to use the Cross entropy first then use PCS-FUDA. It is so weird that the performance lower than before.



---- Replied Message ----
| From | Zangwei ***@***.***> |
| Date | 05/23/2022 14:47 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [zhengzangw/PCS-FUDA] Questions about MIM and K list value (Issue #7) |

My suggestion is to use a large dataset pretrained model (e.g., ImageNet-1k). We find that initial prototype quality is very important, so a pretrained model is needed for a good performance. Another potential solution is to train your model with only cross-entropy first and then with our method.
